CIP-0XXX/README.md

```diff
@@ -0,0 +1,540 @@
+
+### State machine
+
+| Agency ||
+|--------|------| 
+| Client has Agency | StIdle | 
+| Server has Agency | StBusy |
+
+```mermaid
+stateDiagram-v2   
+
+  classDef White fill:white,stroke:white
+  classDef Black fill:white,stroke:black
+  classDef Blue fill:white,stroke:blue
+  classDef Green fill:white,stroke:green
+
+  start:::White --> StIdle:::Blue;
+  StIdle:::Blue --> StBusy:::Green : MsgNextMessage
+  StBusy:::Green --> StIdle:::Blue : MsgHasMessage
+  StBusy:::Green --> StIdle:::Blue : MsgTimeoutMessage
+  StIdle:::Blue --> StDone:::Black : MsgDone
+   
+```
+
+#### Protocol messages
+
+* **MsgNextMessage**: The client asks for the next message.
+* **MsgHasMessage(message)**: The server has received a message.
+* **MsgTimeoutMessage**: The server has not received a message and times out.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgNextMessage |  | StBusy |
+| StBusy | MsgHasMessage | message | StIdle |
+| StBusy | MsgTimeoutMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+# Rationale
+
+## Why are we proposing this CIP?
+
+### For Mithril
+
+* Mithril requires strong network foundations to support interactions between its various nodes:
+
+  * Mithril needs to exist in a decentralized context where multiple aggregators can operate seamlessly and independently.
+  * Mithril needs participation of all or nearly all of the Cardano network SPOs to provide maximal security to the multi-signatures embedded in the certificates.
+  * Creating a separate network would entail significant costs and efforts (there are more than 3,000 SPOs which would need to be connected with resilient and secure network, and much more passive nodes). 
+  * Cardano SPOs need to be vigilant about what other applications run on their block producers and relay nodes. While a separate p2p network could be created, incoming connections must be treated carefully and all the same DoS considerations as with Cardano would need to apply. By standardizing the message diffusion of Mithril in the same way as the Cardano protocol stack, the additional risk of operating Mithril is greatly reduced.
+  * The Cardano network is very efficient for diffusion (e.g. broadcasting) which is precisely what is needed for Mithril.
+  * Mithril signer node needs to run on the same machine as the Cardano block producing node (to access the KES keys). It makes sense to use the same network layer, which will also facilitate a high level of participation.
+
+### For Cardano
+
+* Why would it be great for Cardano to support a decentralized message queue with its network?
+
+  * This is a required feature to make the Cardano ecosystem scalable.
+  * The design is versatile enough to support present and future use cases.
+
+# Information Diffusion Architecture
```

Comment:
```suggestion
### Information Diffusion Architecture
```
This is correctly sequenced with the _Rationale_ but according to CIP convention should also be a part of it  — rather than standing alone as an unconventional section — because it addresses design choices & alternatives.

Reply:
Done in https://github.com/cardano-scaling/CIPs/commit/a3677312b4fa951484ed769f3b69b6e90e93a742